File: tools/credence-lint/corpus/expect-through-accessor/good_uses_stdlib.py
# Role: body
"""Legal: uses stdlib accessors, not structural fields."""
from skin.client import SkinClient


def show_mean(skin: SkinClient, state_id: str) -> float:
    return skin.mean(state_id)


def show_weights(skin: SkinClient, state_id: str) -> list:
    return skin.weights(state_id)

File: tools/credence-lint/corpus/expect-through-accessor/good_uses_stdlib.jl
# Role: brain-side application
# Legal: uses stdlib accessors instead of reading structural fields.

using Credence

function compute_stats(p::BetaPrevision)
    m = mean(p)
    v = variance(p)
    (m, v)
end

function check_event(m::MixtureMeasure, tags::Set{Int})
    probability(m, TagSet(m.space, tags))
end

function show_weights(m::MixtureMeasure)
    w = weights(m)
    println("weights: $w")
end
